configurations/createDockerfile.go

```diff
@@ -0,0 +1,32 @@
+package configurations
+
+import (
+	"log"
+	"os"
+)
+
+// Creates a template and writes it to a new Dockerfile.
+func WriteDockerfile() error {
+	data, err := newDockerFileDataFromYamlFile("code-runner.yaml")
+	if err != nil {
+		return err
+	}
+
+	tmpl := newDockerfileTemplate(data)
+
+	file, err := os.Create("Dockerfile")
+	if err != nil {
+		return err
```

Comment:
resource leak here. The defer function should be placed before any return statement.
In this case, you should place right after line 17.

Reply:
Done